docs/08_jenkins_testing_enviroment.md

```diff
@@ -0,0 +1,130 @@
+# Jenkins testing environment deployment
+
+> Summary of the proposed change
+
+Deploy Jenkins testing environment to be able to integrate it with the Metrics app and be able to have a more real-world environment for testing.
+
+# References
+
+> Link to supporting documentation, GitHub tickets, etc.
+
+[Jenkins](https://jenkins.io/)  
+[Google Cloud Jenkins application](https://github.com/GoogleCloudPlatform/click-to-deploy/blob/master/k8s/jenkins/README.md)
+[Amazon AWS](https://aws.amazon.com/)
+
+# Motivation
+
+> What problem is this project solving?
+
+Creating the Jenkins test environment will provide the data for the Metrics application.
+The Jenkins test environment will give us the ability to test the whole process of importing and displaying the project metrics data.
+
+# Goals
+
+> Identify success metrics and measurable goals.
+
+* Jenkins server is running and available for testing by anyone in the team.
+* Configuration of the Jenkins to run the tests from the test project. 
+
+# Non-Goals
+
+> Identify what's not in scope.
+
+The creation of the test project is not in the scope of this project.
+
+# Design
+
+> Explain and diagram the technical design
+
+`Amazon AWS instance` -> `Jenkins Server`
+
+> Identify risks and edge cases
+
+Because different builds require different environments for Jenkins to run the build 
+(in our case - we need to run the builds with npm, so the Node.js is required), we need to run 
+the builds in a separate docker container and this could require an additional configuration, 
+for example, configuring the docker client, or Jenkins user roles to be able to run the docker container, etc.
+
+# API
+
+> What will the proposed API look like?
+
+The Jenkins API is described in this article: [Jenkins remote access api](https://wiki.jenkins.io/display/JENKINS/Remote+access+API).
```

Comment:
I think there we should describe our suggestion in deploying Jenkins instance to Amazon AWS. For example: 
- `API` section will provide a short annotation about our suggestion. 
- The first subsection (like `Why Amazon AWS?`) will generally describe the `pros` of the suggested approach.
- The second subsection (like `Deploying to Amazon AWS`) will provide more deploying details (i.e. tips).
- The last subsection (like `More configurations`) will generally describe the `cons` of the suggested approach.

All of these points are aimed to remove `Amazon AWS` from the `Alternatives Considered` and `Results` sections since Amazon is not an alternative - this is our choice, and is not the result - the result should match `Goals`. 

What do you think about this?